alacritty/src/event.rs

```diff
@@ -62,16 +61,20 @@ const MAX_SEARCH_WHILE_TYPING: Option<usize> = Some(1000);
 /// Maximum number of search terms stored in the history.
 const MAX_SEARCH_HISTORY_SIZE: usize = 255;
 
-/// Events dispatched through the UI event loop.
+/// Alacritty events.
 #[derive(Debug, Clone)]
-pub enum Event {
-    Terminal(TerminalEvent),
-    DprChanged(f64, (u32, u32)),
-    Scroll(Scroll),
-    ConfigReload(PathBuf),
-    Message(Message),
-    BlinkCursor,
-    SearchNext,
+pub struct Event {
+    /// Limit event to a specific window.
+    window_id: Option<WindowId>,
```

Comment:
This PR currently uses this `Option` to differentiate between "global" and window-specific events. I've also tried to have an enum that differentiates between global and window-specific events, however that would limit each event to one specific purpose.

I believe the only reason right now why we would want the two combined instead of separate enums is that the `Message` event is propagated to all windows without any specific target. This is somewhat unfortunate, the logging in general is a bit of a sore point when it comes to handling of separate issues since it is not window specific (and I don't see a great way to change that).